src/frontend/screens/Audio/AudioAskPermissionBottomSheet.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import {BottomSheetWrapper} from '../../sharedComponents/BottomSheetWrapper';
+import {
+  AppState,
+  AppStateStatus,
+  Linking,
+  StyleSheet,
+  View,
+} from 'react-native';
+import {defineMessages, useIntl} from 'react-intl';
+import AudioPermission from '../../images/observationEdit/AudioPermission.svg';
+import {HeaderText} from '../../sharedComponents/Text/HeaderText';
+import {BodyText} from '../../sharedComponents/Text/BodyText';
+import {Button} from '../../sharedComponents/Button';
+import {Audio} from 'expo-av';
+import {NativeRootNavigationProps} from '../../sharedTypes/navigation';
+
+const m = defineMessages({
+  title: {
+    id: 'screens.AudioPermission.title',
+    defaultMessage: 'Recording Audio with CoMapeo',
+    description: 'Screen title for audio permission screen',
+  },
+  description: {
+    id: 'screens.AudioPermission.description',
+    defaultMessage:
+      'To record audio while using the app and in the background CoMapeo needs to access your microphone. Please enable microphone permissions in your app settings.',
+    description: 'Screen description for audio permission screen',
+  },
+  notNowButtonText: {
+    id: 'screens.AudioPermission.Button.notNow',
+    defaultMessage: 'Not Now',
+    description: 'Screen button text for not granting audio permission',
+  },
+  allowButtonText: {
+    id: 'screens.AudioPermission.Button.allow',
+    defaultMessage: 'Allow',
+    description: 'Screen button text for granting the audio permission',
+  },
+  goToSettingsButtonText: {
+    id: 'screens.AudioPermission.Button.goToSettings',
+    defaultMessage: 'Go to Settings',
+    description:
+      'Screen button text for navigating the user to settings when audio permission is denied',
+  },
+});
+
+export const AudioAskPermissionBottomSheet = ({
+  navigation,
+  route,
+}: NativeRootNavigationProps<'AudioAskPermissionBottomSheet'>) => {
+  const {formatMessage} = useIntl();
+  const {goBack, replace} = navigation;
+  const [permission, setPermission] = React.useState<Audio.PermissionResponse>(
+    route.params.audioPermission,
+  );
+
+  // When the user changes their permission in phone settings and returns to the app,
+  // we need to check for updates and navigate accordingly.
+  // Without this, the app won't detect the permission change.
+  React.useEffect(() => {
```

Comment:
should this be using `useFocusEffect()`? my understanding is that this should only run when this screen is actually focused on (and not hidden behind some other screen, for example).

maybe doesn't have much of difference in impact in practice.